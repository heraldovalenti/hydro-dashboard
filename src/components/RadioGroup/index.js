import React from "react";
import { RadioGroup, FormControlLabel, FormLabel } from "@material-ui/core";
import Radio from "../Radio";

export default props => (
  <>
    {props.legend && <FormLabel component="legend">{props.legend}</FormLabel>}
    <RadioGroup
      className={props.className}
      aria-label={props.name}
      name={props.name}
      value={props.value}
      onChange={event => props.onChange(event.target.value)}
    >
      {props.items.map(item => (
        <FormControlLabel
          key={item.value}
          value={item.value}
          control={<Radio />}
          label={item.label}
        />
      ))}
    </RadioGroup>
  </>
);
