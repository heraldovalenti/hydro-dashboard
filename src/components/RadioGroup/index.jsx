import React from 'react';
import { RadioGroup, FormControlLabel, FormLabel } from '@mui/material';
import Radio from '../Radio';

export default (props) => (
  <>
    {props.legend && <FormLabel component="legend">{props.legend}</FormLabel>}
    <RadioGroup
      className={props.className}
      aria-label={props.name}
      name={props.name}
      value={props.value}
      onChange={(event) => props.onChange(event.target.value)}
    >
      {props.items.map((item, index) => (
        <FormControlLabel
          key={`${item.value}_${index}`}
          value={item.value}
          control={<Radio />}
          label={item.label}
          disabled={props.disabled}
        />
      ))}
    </RadioGroup>
  </>
);
