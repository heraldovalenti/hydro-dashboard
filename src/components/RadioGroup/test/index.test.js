import React from "react";
import RadioGroup from "../";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<RadioGroup/>", () => {
  it("renders no items", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RadioGroup items={[]} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it("renders several items", () => {
    const items = [
      { value: 1, label: "one" },
      { value: 2, label: "two" }
    ];
    const renderer = new ShallowRenderer();
    renderer.render(<RadioGroup items={items} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
