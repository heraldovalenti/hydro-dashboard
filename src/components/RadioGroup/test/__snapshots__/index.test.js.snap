// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup/> renders no items 1`] = `
<React.Fragment>
  <ForwardRef(RadioGroup)
    onChange={[Function]}
  />
</React.Fragment>
`;

exports[`<RadioGroup/> renders several items 1`] = `
<React.Fragment>
  <ForwardRef(RadioGroup)
    onChange={[Function]}
  >
    <ForwardRef(WithStyles)
      control={<ForwardRef(WithStyles) />}
      label="one"
      value={1}
    />
    <ForwardRef(WithStyles)
      control={<ForwardRef(WithStyles) />}
      label="two"
      value={2}
    />
  </ForwardRef(RadioGroup)>
</React.Fragment>
`;
